### PR TITLE
Map utilities window

### PR DESCRIPTION
**Pull request by Garanas**
Replaces the marker utilities window introduced by #3387 .

Given that:
 - There is an AI in-game
 - Cheating is enabled
 - There is a single human player ingame

Then this window can:

image

Visualize the iMAP grid used by the AI to identify threat values. You can toggle each threat value individually, allowing you to see what the AI is seeing.

image

Toggle each marker type available in the game, allowing you to see what the AI is using.

image

Perform a resource check to check if you can:
 - Build an extractor at mass maskers
 - Build a hydrocarbon at hydro markers
 - Ring the extractor with storages (as done by the ringing feature introduced by #3484)
 - Ring the extractor with fabricators (as done by the ringing feature introduced by #3484)
 
 All in all, a useful tool for map makers and AI developers. It requires a few additional checks before merging:
  - Anti cheat measurements, add additional conditions to the sim callbacks
  - Documentation of structure and commenting in general
  
 Also creates the foundation of how I intend to format the architecture of the profiler in #3674 .

### CODE REVIEW

File: lua/sim/MapUtilities.lua [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+                            a[2] = GetTerrainHeight(cx, cz)
+                            a[3] = cz
+
+                            -- draw individual threat values of cell
+                            for k, info in threatIdentifiers do 
+
+                                -- DrawCircle(a, 0.25 * sqrt(1600), "000000")
+
+                                if threadRendering[info.identifier] then 
+                                    local threat = brain:GetThreatAtPosition(a, 1, true, info.identifier)
```
Comment (Sprouto) [inline lua/sim/MapUtilities.lua:207]: This command should be issued with a 0 value for rings - not 1 - 1 will pick up threat from adjacent IMAP blocks as part of it's total. 0 will restrict it to just the threat in this block.

Comment (Garanas) [reply lua/sim/MapUtilities.lua:207]: Ah, I thought that was a radius in terms of distance on the map. That explains why we have 3x3 blocks 😄

**Comment by HUSSAR-mtrela** [comment]
I really like this new UI for visualizes what AI is seeing and planning.

**Comment by Garanas** [comment]
It is almost complete - just a last few tweaks 😄

**Pull request merged**